pool/options.go

```diff
@@ -0,0 +1,83 @@
+package pool
+
+import (
+	"errors"
+	"fmt"
+	"path"
+	"time"
+
+	"github.com/libp2p/go-libp2p"
+	"github.com/libp2p/go-libp2p/core/host"
+)
+
+type (
+	Option  func(*options) error
+	options struct {
+		h                host.Host
+		name             string
+		topicName        string
+		announceInterval time.Duration
+	}
+)
+
+func newOptions(o ...Option) (*options, error) {
+	opts := options{
+		announceInterval: 5 * time.Second,
+	}
+	for _, apply := range o {
+		if err := apply(&opts); err != nil {
+			return nil, err
+		}
+	}
+	if opts.name == "" {
+		return nil, errors.New("pool name must be specified")
+	}
+	if opts.topicName == "" {
+		opts.topicName = fmt.Sprintf("/fx.land/pools/%s", path.Clean(opts.name))
```

Comment:
Not that important but what about giving it a name like /explore.fula/pools/%s ?
We also own fula tld on unstoppable

Reply:
Sure